Lib/glyphsLib/builder/glyph.py

```diff
@@ -40,23 +40,26 @@ def to_ufo_glyph(self, ufo_glyph, layer, glyph):
     last_change = glyph.lastChange
     if last_change is not None:
         ufo_glyph.lib[GLYPHLIB_PREFIX + 'lastChange'] = to_ufo_time(last_change)
+
     color_index = glyph.color
     if color_index is not None:
-        color_tuple = None
+        # On writing a file, Glyphs clamps the color values to of a color
+        # tuple to uint8 range and sets the alpha channel to 1. Try to do
+        # the right thing for alpha instead. .3f is enough precision to
+        # round-trip uint8 to float losslessly.
+        # https://github.com/unified-font-object/ufo-spec/issues/61#issuecomment-389759127
         if isinstance(color_index, list):
-            if not all(i in range(0, 256) for i in color_index):
-                logger.warning('Invalid color tuple {} for glyph {}. '
-                               'Values must be in range 0-255'.format(
-                                   color_index, glyph.name))
-            else:
-                color_tuple = ','.join('{0:.4f}'.format(i/255) if i in range(1, 255) else str(i//255) for i in color_index)
-        elif isinstance(color_index, int) and color_index in range(len(GLYPHS_COLORS)):
-            color_tuple = GLYPHS_COLORS[color_index]
+            color_values_clamped = tuple(
+                max(min(abs(int(v)), 255), 0) for v in color_index[:4])
```

Comment:
I don't understand why you have to clamp here. Raising ValueError when the color integer is not in range 0..255 was the correct thing to do here. The argument that Glyphs.app doesn't crash if a user attempts to set a color integer which is not in range to me does not apply here. In the case of glyphsLib, receiving invalid data must be  shouted out loud, not silently fixed as you do here.

why do you take the first four items with the slice `color_index[:4]`?. Should you not raise if the color tuple does not contain 4 items, and is that even possible? I believe it always contains 4 items in the glyphs source file, and if it doesn't, well it's an error.